Assets/Scripts/VR/VRControllers/VRControllerManager.cs

```diff
@@ -0,0 +1,304 @@
+
+        public enum ControllerModel { Index, Quest, Quest2 }
+
+
+        internal void InitializeControllers(string name)
+        {
+            switch (name)
+            {
+                case "Index Controller OpenXR": InitializeControllers(ControllerModel.Index); break;
+                case "Oculus Touch Controller OpenXR": InitializeOculusController(GetHMD()); break;
+                default: InitializeControllers(ControllerModel.Quest); break;
+            }
+        }
+
+        internal void InitializeOculusController(string name)
+        {
+            switch (name)
+            {
+                case "Oculus Rift S": InitializeControllers(ControllerModel.Quest); break;
+                case "Oculus Quest": InitializeControllers(ControllerModel.Quest); break;
+                case "Oculus Quest2": InitializeControllers(ControllerModel.Quest2); break;
+                default: InitializeControllers(ControllerModel.Quest); break;
+            }
+        }
+
+        public void InitializeControllers(ControllerModel model)
+        {
+            if (null != rightController) rightController.controllerTransform.gameObject.SetActive(false);
+            if (null != leftController) leftController.controllerTransform.gameObject.SetActive(false);
+            if (null != inverseRightController) inverseRightController.controllerTransform.gameObject.SetActive(false);
+            if (null != inverseLeftController) inverseLeftController.controllerTransform.gameObject.SetActive(false);
+            switch (model)
+            {
+                case ControllerModel.Index:
+                    GetIndexControllersValues();
+                    break;
+                case ControllerModel.Quest:
+                    GetQuestControllersValues();
+                    break;
+                case ControllerModel.Quest2:
+                    GetQuest2ControllersValues();
+                    break;
+            }
+            SetRightHanded(GlobalState.Settings.rightHanded);
+        }
+
+        private static string GetHMD()
+        {
+            string exe = "VRtistOpenXRHMD.exe";
+#if UNITY_EDITOR
+            string path = "./Data/bin";
+#else
+            string path = "./VRtist_Data/bin";
+#endif
+            string fullpath = System.IO.Path.GetFullPath(path) + "/" + exe;
+            System.Diagnostics.ProcessStartInfo info = new System.Diagnostics.ProcessStartInfo(fullpath);
+            info.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+            info.CreateNoWindow = true;
+            info.RedirectStandardOutput = true;
+            info.UseShellExecute = false;
+            System.Diagnostics.Process process = System.Diagnostics.Process.Start(info);
+            process.WaitForExit();
+            return process.StandardOutput.ReadLine();
+        }
+
+        public Transform GetPrimaryControllerTransform()
+        {
+            if (rightController == null) InitializeControllers(ControllerModel.Quest);
```

Comment:
InitializeControllers should be called somewhere else or also in the GetSecondaryControllerTransform function.